pkg/mobile/bindableServices.go

```diff
@@ -0,0 +1,189 @@
+)
+
+type BindableMobileServiceCRUDLImpl struct {
+	scClient         scv1beta1.ServicecatalogV1beta1Interface
+	mobileClientRepo MobileClientRepo
+	secretsCRUDL     SecretsCRUDL
+}
+
+func NewServiceBindingLister(scClient scv1beta1.ServicecatalogV1beta1Interface, mobileClientRepo MobileClientRepo, secretsCRUDL SecretsCRUDL) *BindableMobileServiceCRUDLImpl {
+	return &BindableMobileServiceCRUDLImpl{
+		scClient:         scClient,
+		mobileClientRepo: mobileClientRepo,
+		secretsCRUDL:     secretsCRUDL,
+	}
+}
+
+func (lister *BindableMobileServiceCRUDLImpl) Create(namespace string, binding *ServiceBinding, formData map[string]interface{}) (*ServiceBinding, error) {
+
+	bindingsApi := lister.scClient.ServiceBindings(namespace)
+	binding2, err := bindingsApi.Create(binding)
+	if err != nil {
+		return nil, err
+	}
+
+	parametersSecret := makeParametersSecret(binding, binding2, formData)
+	_, err = lister.secretsCRUDL.Create(namespace, parametersSecret)
+
+	if err != nil {
+		return nil, err
+	}
+	return binding2, nil
+
+}
+
+func (lister *BindableMobileServiceCRUDLImpl) Delete(namespace string, bindingName string) error {
+	return lister.scClient.ServiceBindings(namespace).Delete(bindingName, &v1.DeleteOptions{})
+}
+
+func (lister *BindableMobileServiceCRUDLImpl) List(namespace string, mobileClientName string) (*BindableMobileServiceList, error) {
+
+	listOpts := v1.ListOptions{}
+
+	getOpts := v1.GetOptions{}
+
+	serviceBindingList := BindableMobileServiceList{}
+
+	serviceBindings, err := lister.scClient.ServiceBindings(namespace).List(listOpts)
+
+	if err != nil {
+		return nil, err
+	}
+
+	serviceInstances, err := lister.scClient.ServiceInstances(namespace).List(listOpts)
+
+	if err != nil {
+		return nil, err
+	}
+
+	for _, serviceInstance := range serviceInstances.Items {
+
+		bindableService := BindableMobileService{}
+
+		csc, err := lister.scClient.ClusterServiceClasses().Get(serviceInstance.Spec.ClusterServiceClassRef.Name, getOpts)
+
+		if err != nil {
+			return nil, err
+		}
+
+		if csc.Spec.ExternalMetadata != nil {
+
+			var x interface{}
+
+			json.Unmarshal(csc.Spec.ExternalMetadata.Raw, &x)
+			m := x.(map[string]interface{})
+
+			bindableService.Name = m["displayName"].(string)
+
+			if m["imageUrl"] != nil {
+				bindableService.ImageURL = m["imageUrl"].(string)
+			} else {
+				if m["console.openshift.io/iconClass"] != nil {
+					bindableService.IconClass = m["console.openshift.io/iconClass"].(string)
+				} else {
+					bindableService.ImageURL = "https://avatars1.githubusercontent.com/u/3380462?s=200&v=4"
+				}
+			}
+
+		}
+
+		servicePlans, err := lister.scClient.ClusterServicePlans().List(listOpts)
+
+		if err != nil {
+			return nil, err
+		}
+		var servicePlan ServicePlan
+
+		for _, sb := range servicePlans.Items {
+			if sb.Spec.ClusterServiceClassRef.Name == csc.Name {
+				servicePlan = sb
+			}
+		}
+
+		var serviceBinding ServiceBinding
+
+		for _, sb := range serviceBindings.Items {
```

Comment:
similarly, I think this can be moved to its own function to reduce the size of this one